#!/usr/bin/env python3 

PACKAGE = "param_examples"

from dynamic_reconfigure.parameter_generator_catkin import *


gen = ParameterGenerator()


gen.add('int_param',int_t,0,"Intger param",5,-10,15)
gen.add('str_param',str_t,0,"String param","hello")
gen.add('bool_param',bool_t,0,"Bool Param",True)

size_enu_matrix = gen.enum([gen.const("low",int_t,0,"lowest value"),
                            gen.const("medium",int_t,5,"medium value"),
                            gen.const("high",int_t,10,"high value")],"An enum for setting")

gen.add("size",int_t,0,"size parameter via enu",0,0,10,edit_method=size_enu_matrix)

exit(gen.generate(PACKAGE,"dynamic_param_examples","dynamic_param"))